components/common/src/test/java/com/hotels/styx/config/validator/SchemaTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.config.validator;
+
+import com.hotels.styx.config.validator.Schema.Field;
+import org.testng.annotations.Test;
+
+import static com.hotels.styx.config.validator.ObjectValidator.schema;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.is;
+
+public class SchemaTest {
```

Comment:
There are many test names that do not conform to our naming conventions. The name should describe what happens and under what condition.